src/Compilers/Core/Portable/DiagnosticAnalyzer/Suppression.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    /// <summary>
+    /// Programmatic suppression of a <see cref="Diagnostic"/> by a <see cref="DiagnosticSuppressor"/>.
+    /// </summary>
+    public struct Suppression
+    {
+        private Suppression(SuppressionDescriptor descriptor, Diagnostic suppressedDiagnostic)
```

Comment:
>suppressedDiagnostic [](start = 73, length = 20)

What is the expected relationship of this value with ```DiagnosticWithProgrammaticSuppression```? Can/must it be a ```DiagnosticWithProgrammaticSuppression``` instance?

Reply:
`Suppression` is the diagnostic suppressor author facing API to report suppressions. The analyzer driver then takes the reported suppressions from all suppressors and creates a single ProgrammaticSuppressionInfo per diagnostic and creates a new `DiagnosticWithProgrammaticSuppression`. Basically, `Suppression` is a concept for each suppression author, and the `ProgrammaticSuppressionInfo` and `DiagnosticWithProgrammaticSuppression` is the output concept for diagnostic consumers/hosts.

Reply:
> Suppression is the diagnostic suppressor author facing API to report suppressions. The analyzer driver then takes the reported suppressions from all suppressors and creates a single ProgrammaticSuppressionInfo per diagnostic and creates a new DiagnosticWithProgrammaticSuppression. Basically, Suppression is a concept for each suppression author, and the ProgrammaticSuppressionInfo and DiagnosticWithProgrammaticSuppression is the output concept for diagnostic consumers/hosts.

I am assuming then that suppressedDiagnostic cannot be a DiagnosticWithProgrammaticSuppression. Consider adding an assert for that.


---
In reply to: [292213060](https://github.com/dotnet/roslyn/pull/36067#discussion_r292213060) [](ancestors = 292213060)

Reply:
Yes, that is correct. I have added an assert.